packages/flutter_tools/test/general.shard/web/devices_test.dart

```diff
@@ -138,6 +138,22 @@ void main() {
       contains(isA<GoogleChromeDevice>()));
   });
 
+  testWithoutContext('Edge device is listed when Edge can be run', () async {
+    final WebDevices webDevices = WebDevices(
+      featureFlags: TestFeatureFlags(isWebEnabled: true),
+      fileSystem: MemoryFileSystem.test(),
+      logger: BufferLogger.test(),
+      platform: FakePlatform(
+        operatingSystem: 'linux',
+        environment: <String, String>{}
+      ),
+      processManager:  FakeProcessManager.any(),
+    );
+
+    expect(await webDevices.pollingGetDevices(),
+      contains(isA<MicrosoftEdgeDevice>()));
```

Comment:
I'm surprised that this passes. The `findEdgeExecutable` function specifically checks that `platform.isWindow`, and this test uses 'linux' as the OS.

Reply:
Edge is supported on all platforms, it just matters whether the executable can be located.

Reply:
According to https://github.com/flutter/flutter/blob/0830c75c59b462975dbfb8da642ed8a2a3ff4550/packages/flutter_tools/lib/src/web/chrome.dart#L97 it won't be found, unless you explicitly specify the `EDGE_ENVIRONMENT` variable. Is that what you mean by "executable can be located"?

Reply:
Fixed to only check on Windows